src/executables/util_cs.c

```diff
@@ -3993,3 +3993,255 @@ tde (UTIL_FUNCTION_ARG * arg)
+    {
+      fprintf (stderr, "wrong dbname\n");
+      goto error_exit;
+    }
+
+  /* create table list */
+  /* class existence and classoid will be found at server side. if is checked at utility side, it needs addtional access to the server through locator */
+  darray = da_create (num_tables, SM_MAX_IDENTIFIER_LENGTH);
+  if (darray == NULL)
+    {
+      perror ("calloc");
+      util_log_write_errid (MSGCAT_UTIL_GENERIC_NO_MEM);
+      goto error_exit;
+    }
+
+  for (i = 0; i < num_tables; i++)
+    {
+      table_name = utility_get_option_string_value (arg_map, OPTION_STRING_TABLE, i + 1);
+      if (table_name == NULL)
+	{
+	  continue;
+	}
+
+      strncpy_bufsize (table_name_buf, table_name);
+      if (da_add (darray, table_name_buf) != NO_ERROR)
+	{
+	  util_log_write_errid (MSGCAT_UTIL_GENERIC_NO_MEM);
+	  perror ("calloc");
+	  goto error_exit;
+	}
+    }
+
+  /* start date check */
+  if (start_date != NULL && strlen (start_date) > 0)
+    {
+      start_time = parse_date_string_to_time (start_date);
+      if (start_time == 0)
+	{
+	  fprintf (stderr, "start-date error : follow  DATE format (DD-MM-YYYY:hh:MM:ss)\n");
+	  goto error_exit;
+	}
+    }
+
+  /* end date check */
+  if (end_date != NULL && strlen (end_date) > 0)
+    {
+      end_time = parse_date_string_to_time (end_date);
+      if (end_time == 0)
+	{
+	  fprintf (stderr, "end-date error : follow  DATE format (DD-MM-YYYY:hh:MM:ss)\n");
+	  goto error_exit;
+	}
+    }
+
+  /* start time and end time setting.
+   * 1. 10 minutes before end time, if only end time is specified
+   * 2. 10 minutes after start time, if only start time is specified
+   * 3. if no time is specified, then get current execution time for end time */
+
+  if (start_time == 0 && end_time != 0)
+    {
+      start_time = end_time - 600;
+    }
+  else if (start_time != 0 && end_time == 0)
+    {
+      end_time = start_time + 600;
+    }
+  else if (start_time == 0 && end_time == 0)
+    {
+      end_time = time (NULL);
+      start_time = end_time - 600;
+    }
+
+  /* 1. start time < end time
+   * 2. start time is required to be set after db_creation time (server side check)
+   * 3. start time, and end time are required to be set within the log volume range (server side check) */
+  if (start_time > end_time)
+    {
+      fprintf (stderr, "start time(%lld) is larger than end time(%lld)\n", start_time, end_time);
+      goto error_exit;
+    }
+
+  /* output file check */
+  if (output_file == NULL)
+    {
+      outfp = stdout;
+    }
+  else
+    {
+      outfp = fopen (output_file, "w");
+      if (outfp == NULL)
+	{
+	  fprintf (stderr, "can not open output file\n");
+	  goto error_exit;
+	}
+    }
+
+  /* error message log file */
+  snprintf (er_msg_file, sizeof (er_msg_file) - 1, "%s_%s.err", database_name, arg->command_name);
+  er_init (er_msg_file, ER_NEVER_EXIT);
+
+  db_set_client_type (DB_CLIENT_TYPE_ADMIN_UTILITY);
+  if (db_login ("DBA", dba_password) != NO_ERROR)
```

Comment:
```suggestion
  if (db_login (user, dba_password) != NO_ERROR)
```

Reply:
And how do you check whether the user has DBA privileges? It looks like you need to use the au_is_dba_group_member() function, but you don't use it, so I leave a question.

Reply:
'user' option is not a user that execute a flashback utility. It is a filter option that extract SQL logs executed by a transaction 'user'. 
All the other utilities which need a dba privilege doesn't specify the user who executed.

Reply:
@joohok Oh. I'm confused. I hope you can answer the second question as well.

Reply:
Utility will be executed as a "DBA" user, so I think it doesn't need to check the dba privilege. 
How do you think?

Reply:
@joohok I got it.